### PR TITLE
feat(artifacts): Reference ExpectedArtifacts by id.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

**Comment by jtk54** [comment]
Depends on https://github.com/spinnaker/echo/pull/194 and https://github.com/spinnaker/orca/pull/1763.

File: app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.html [hunk 1]
```diff
@@ -116,9 +116,8 @@
         <div class="col-md-7">
           <ui-select ng-model="command.expectedArtifact"
                      class="form-control input-sm">
-            <ui-select-match>{{ basicSettingsCtrl.summarizeExpectedArtifact(command.expectedArtifact) }}
-            </ui-select-match>
-            <ui-select-choices repeat="expected in command.backingData.expectedArtifacts">
+            <ui-select-match>{{  basicSettingsCtrl.summarizeExpectedArtifact($select.selected) }}</ui-select-match>
+            <ui-select-choices repeat="expected.id as expected in command.backingData.expectedArtifacts">
```
Comment (lwander) [inline app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.html]: How are these surfaced? Does it look like uuids or like the summary before?

Comment (jtk54) [reply app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.html]: Summary like before:

image

File: app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.html [hunk 2]
```diff
@@ -116,9 +116,8 @@
         <div class="col-md-7">
           <ui-select ng-model="command.expectedArtifact"
                      class="form-control input-sm">
-            <ui-select-match>{{ basicSettingsCtrl.summarizeExpectedArtifact(command.expectedArtifact) }}
-            </ui-select-match>
-            <ui-select-choices repeat="expected in command.backingData.expectedArtifacts">
+            <ui-select-match>{{  basicSettingsCtrl.summarizeExpectedArtifact($select.selected) }}</ui-select-match>
```
Comment (danielpeach) [inline app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.html]: Extra space `{{  basicSettingsCtrl`

**Pull request merged**